plugin.video.nhlgcl/resources/lib/globals.py

```diff
@@ -0,0 +1,644 @@
+import sys
+import re, os, time
+import calendar
+import pytz
+import urllib
+import requests
+import json
+import http
+import time
+from bs4 import BeautifulSoup
+from datetime import date, datetime, timedelta
+from kodi_six import xbmc, xbmcplugin, xbmcgui, xbmcaddon
+
+if sys.version_info[0] > 2:
+    urllib = urllib.parse
+    cookielib = http.cookiejar
+
+addon_handle = int(sys.argv[1])
+
+# Addon Info
+ADDON = xbmcaddon.Addon()
+ADDON_ID = ADDON.getAddonInfo('id')
+ADDON_VERSION = ADDON.getAddonInfo('version')
+ADDON_PATH = xbmc.translatePath(ADDON.getAddonInfo('path'))
+ADDON_PATH_PROFILE = xbmc.translatePath(ADDON.getAddonInfo('profile'))
+XBMC_VERSION = float(re.findall(r'\d{2}\.\d{1}', xbmc.getInfoLabel("System.BuildVersion"))[0])
+LOCAL_STRING = ADDON.getLocalizedString
+
+# Settings
+settings = xbmcaddon.Addon()
+CDN = str(settings.getSetting(id="cdn"))
```

Comment:
matrix python api contains a lot of new methods you can use to get the respective setting with a pre-defined type (and thus avoid the conversion on the python side). Please consider to use them in future updates

Reply:
Not sure what I'm doing wrong but I can not get past this error when using the new method on the latest nightly.
```
- NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
Error Type: <class 'TypeError'>
Error Contents: Invalid setting type
Traceback (most recent call last):
    File "C:\Users\ejg\AppData\Roaming\Kodi\addons\plugin.video.nhlgcl\addon.py", line 1, in <module>
    from resources.lib.nhl_tv import *
    File "C:\Users\ejg\AppData\Roaming\Kodi\addons\plugin.video.nhlgcl\resources\lib\nhl_tv.py", line 1, in <module>
    from resources.lib.globals import *
    File "C:\Users\ejg\AppData\Roaming\Kodi\addons\plugin.video.nhlgcl\resources\lib\globals.py", line 30, in <module>
    CDN = settings.getSettingString("cdn")
TypeError: Invalid setting type
-->End of Python script error report<--
```

Reply:
It's working fine for me but I'm running current master. "cdn" setting is a `CSettingString` so it complies to `getSettingString`. But it is fine as is